Move documentation from README into the manifests

stevenpost opened the pull request:
Documentation can be more easily generated using docstrings.

This was cherry-picked from @witjoh's WIP https://github.com/voxpupuli/puppet-mongodb/pull/677.

h-haaks commented:
I would remove the reference section in README.mb as well

h-haaks commented on manifests/globals.pp:
```diff
@@ -1,29 +1,118 @@
+#
+#   class {'mongodb::globals':
+#      manage_package_repo => true,
+#      repo_location       => 'http://example.com/repo'
```
I removed this example from README.md see my aguments in #707

h-haaks commented on manifests/globals.pp:
```diff
@@ -1,29 +1,118 @@
+#
+#   class {'mongodb::globals':
+#     manage_package_repo => true,
+#     version             => '3.6',
```
I don't think setting version to major.minor is working.
It will try to install packages with this version and they don't exist.

Change to repo_version?

h-haaks commented:
@stevenpost I'm woriking on a PR to move from params to hiera. 
Could you please look at my comments so that we get this merged before I change to many classes?

stevenpost commented:
> I would remove the reference section in README.md as well

done

h-haaks commented:
Did you use the 'strings:generate:reference' rake task to update Reference.md?

stevenpost commented:
> Did you use the 'strings:generate:reference' rake task to update Reference.md?

I used the command `puppet strings generate --format markdown` instead. The rake task gives me errors, I should really look into that. I probably just need to change something on my local bundle config to include development.

h-haaks commented:
The example with repo version both in readme and globals class uses 3.6.

Since 5.0 is default now and only 4.4 really is an option, should we change to 4.4 and set the title to something like
'Specify MongoDB repository version' ?

This applies to enterprise as well as community repos.

h-haaks commented:
Jepp, hence my thumbs up :)

stevenpost commented:
> The example with repo version both in readme and globals class uses 3.6.
> 
> Since 5.0 is default now and only 4.4 really is an option, should we change to 4.4 and set the title to something like 'Specify MongoDB repository version' ?
> 
> This applies to enterprise as well as community repos.

I added a separate example for installing from enterprise, just to make it clear. Added some style fixes as well.

h-haaks commented:
`PS: the module could use a modulesync, my PR on the CONTRIBUTING template got in and the old commands no longer work.`
Or we could just create a PR that reverts that change.

stevenpost commented:
> `PS: the module could use a modulesync, my PR on the CONTRIBUTING template got in and the old commands no longer work.` Or we could just create a PR that reverts that change.

Revert what? I'm referring to https://github.com/voxpupuli/modulesync_config/pull/881

h-haaks commented:
Hmm, I didn't notice, someone just stepped in two days ago and released 5.0.0 ...

h-haaks commented:
> > `PS: the module could use a modulesync, my PR on the CONTRIBUTING template got in and the old commands no longer work.` Or we could just create a PR that reverts that change.
> 
> Revert what? I'm referring to [voxpupuli/modulesync_config#881](https://github.com/voxpupuli/modulesync_config/pull/881)

Ah I see. But there is no new release of module_sync yet.
It's usually pushed to all modules when released I think.
